Devices enumeration: workaround for MacOS

andreibancioiu opened the pull request:
## How to test

Clone the repository and, while having a device connected  (with the MultiversX app open - as an example), run the following command:

```
python3 -m ledgercomm.cli.send --hid stdin <<<ed02000000
```

It should work with no error (on all operating systems). Output should be similar to:

```
ledgercomm - DEBUG - => ed02000000
ledgercomm - DEBUG - <= 0102000100140000000000000000 9000
```

fbeutin-ledger commented on ledgercomm/interfaces/hid_device.py:
```diff
@@ -35,8 +43,7 @@ class HID(Comm):
     def __init__(self, vendor_id: int = 0x2C97) -> None:
```
Use the new DEFAULT_VENDOR_ID

andreibancioiu replied:
Thank you, fixed.

lpascal-ledger commented:
Thanks for the PR @andreibancioiu! We'll need some time to test it on a proper device, but it'll definitely be integrated then.

fbeutin-ledger commented on ledgercomm/interfaces/hid_device.py:
```diff
@@ -53,13 +63,45 @@ def open(self) -> None:
+
+        # On MacOS, "interface_number" is not a reliable filtering criteria, so we
+        # also filter by "usage_page".
+        devices = [device for device in devices if device.get(
+            "usage_page") == MACOS_USAGE_PAGE]
```
Hello, this fails on Linux if two devices are plugged. I've fixed it with:
```suggestion
        mac_filtered_devices = [device for device in devices if device.get(
             "usage_page") == MACOS_USAGE_PAGE]
        if len(mac_filtered_devices) != 0:
            # If we are on a MAC device, we keep the new filtered list
            devices = mac_filtered_devices
```

fbeutin-ledger commented:
Hello @andreibancioiu , sorry about this the CI is failing.. Can you take a quick look please? I'll do it next week otherwise

andreibancioiu commented:
> Hello @andreibancioiu , sorry about this the CI is failing.. Can you take a quick look please? I'll do it next week otherwise

Hello @fbeutin-ledger,

Thanks for the note :pray: 

Fixed the linting issues:
 - I was using the deprecated `LOG.warn()` instead `LOG.warning()`
 - I was raising the base `Exception` at some point. Fixed by defining a custom error: `CannotFindDeviceError`

fbeutin-ledger commented:
Merged into LedgerHQ:fbe/merge_mac_support
I'll fix the lint there

fbeutin-ledger commented:
https://github.com/LedgerHQ/ledgercomm/pull/14
